### PR TITLE
Domain enhancements

### PR DESCRIPTION
**Pull request by jschueller**
- Build DomainUnion/Intersection/DisjunctiveUnion from a collection of domains instead of 2 instances:
```
union = ot.DomainUnion([d1, d2])
intersec = ot.DomainIntersection([d1, d2])
dunion = DomainDisjunctiveUnion([d1, d2])
```
the old `(Domain, Domain)` ctors are deprecated

- implement DomainDisjunctiveUnion::computeDistance:
```
interval1 = ot.Interval(-0.4, 0.0)
interval2 = ot.Interval(1.0, 1.3)
domain = ot.DomainDisjunctiveUnion([interval1, interval2])
x = [0.1]*2
print(domain.computeDistance(x)
```

- new Domain::computeNearestPoints(Point)->Sample implemented for Interval/DomainUnion/DomainDisjunctiveUnion
```
dim = 2
cube = ot.Interval([0.0] * dim, [2.0] * dim)
cube2 = ot.Interval([4.0] * dim, [6.0] * dim)
domain3 = ot.DomainUnion([cube, cube2])
for x in [-1, 3,]:
    for y in [-1, 3]:
        p = [x, y]
        points = domain3.computeNearestPoints(p)
        print(f"p={p} points={points}")
```

### CODE REVIEW

File: python/src/DomainImplementation_doc.i.in [hunk 1]
```diff
@@ -63,6 +63,24 @@ OT_Domain_computeDistance_doc
 
 // ---------------------------------------------------------------------
 
+%define OT_Domain_computeNearestPoints_doc
+"Compute the nearest points to the domain.
```
Comment (josephmure) [inline python/src/DomainImplementation_doc.i.in:67]: It's the reverse!
```suggestion
"Compute the points in (or at the boundary of) the domain which are closest to a given point.
```

**Review by josephmure — APPROVED** [review]
Good to go IMO after this small correction.

File: python/src/DomainImplementation_doc.i.in (hunk 1 above)
Comment (jschueller) [reply python/src/DomainImplementation_doc.i.in]: !

**Comment by jschueller** [comment]
postponing the computeNearestPoints part for now after @regislebrun comment

**Review by adutfoy — APPROVED** [review]

**Review by regislebrun — APPROVED** [review]

**Pull request merged**